api/v1alpha1/dnsrecord_types.go

```diff
@@ -103,7 +103,8 @@ type DNSRecordSpec struct {
 	// +kubebuilder:validation:Pattern=`^(?:[\w\-.~:\/?#[\]@!$&'()*+,;=]+)\.(?:[\w\-.~:\/?#[\]@!$&'()*+,;=]+)$`
 	RootHost string `json:"rootHost"`
 
-	// providerRef is a reference to a provider secret.
+	// ProviderRef is a reference to a provider secret.
```

Comment:
Separate task, but we should go through all these and make them consistent one way or the other, we have a mix at the moment. Probably caused by the fact these get used in the CRD docs so the capitalization of the field name looks wrong there, although it's correct here.